File: atvd presenças/2.py
soma = 0
contador = 0
maior = float('-inf')
menor = float('inf')
soma_par=0
soma_impar=0
faixas = {
    "faixa 1": {"contagem": 0, "soma": 0},
    "faixa 2": {"contagem": 0, "soma": 0},
    "faixa 3": {"contagem": 0, "soma": 0},
    "faixa 4": {"contagem": 0, "soma": 0},
    "faixa 5": {"contagem": 0, "soma": 0}
}
pares=[]
impares=[]
while True:
    n = input("Digite um valor qualquer a ser calculado, quando desejar parar e calcular, digite um caractere:")
    try:
        n = float(n)
        soma += n
        contador += 1
        if n < menor:
            menor = n
        if n > maior:
            maior = n

        if n%2==0:
            pares.append(n)
            soma_par+=n
        else:
            impares.append(n)
            soma_impar+=n

        if n < 0:
            faixa = "faixa 1"
        elif 0 <= n < 15:
            faixa = "faixa 2"
        elif 15 <= n < 100:
            faixa = "faixa 3"
        elif 101 <= n < 1000:
            faixa = "faixa 5"
        elif n >= 1000:
            faixa = "faixa 4"
        else:
            continue  # Ignorar números fora das faixas definidas

        faixas[faixa]["contagem"] += 1
        faixas[faixa]["soma"] += n
    except ValueError:
        break

if contador > 0:
    media = soma / contador
    print("A média dos valores inseridos é: {:.2f}".format(media))
    print("O menor valor é: ", menor)
    print("O maior valor é: ", maior)
    print("valores pares:",pares,"a soma é:",soma_par)
    print("valores impares:", impares,"a soma é:", soma_impar)
else:
    print("Nenhum número foi inserido.")

for faixa_nome, dados in faixas.items():
    print(faixa_nome + " - Total de elementos: " + str(dados['contagem']) + ", Total da faixa: {:.2f}".format(dados['soma']))

File: atvd presenças/2.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    float soma = 0, maior = -1.0 / 0.0, menor = 1.0 / 0.0; // -inf e inf
    float soma_par = 0, soma_impar = 0;
    int contador = 0, indice_par = 0, indice_impar = 0;
    int pares[1000], impares[1000];  // Arrays para armazenar números inteiros
    int faixas_contagem[5] = {0, 0, 0, 0, 0};
    float faixas_soma[5] = {0, 0, 0, 0, 0};
    char input[50];
    float n;
    
    while (1) {
        printf("Digite um valor qualquer a ser calculado, quando desejar parar e calcular, digite um caractere: ");
        if (scanf("%s", input) != 1) break;

        if (sscanf(input, "%f", &n) != 1) {
            break; // Se o input não for um número, sai do loop
        }

        soma += n;
        contador++;

        if (n < menor) menor = n;
        if (n > maior) maior = n;

        // Verificar se o número é um inteiro antes de verificar paridade
        if ((int)n == n) {  // Verifica se o número é inteiro
            if ((int)n % 2 == 0) {
                pares[indice_par] = (int)n;  // Armazenar como inteiro
                soma_par += (int)n;  // Soma como inteiro
                indice_par++;
            } else {
                impares[indice_impar] = (int)n;  // Armazenar como inteiro
                soma_impar += (int)n;  // Soma como inteiro
                indice_impar++;
            }
        }

        // Faixas
        if (n < 0) {
            faixas_contagem[0]++;
            faixas_soma[0] += n;
        } else if (n < 15) {
            faixas_contagem[1]++;
            faixas_soma[1] += n;
        } else if (n < 100) {
            faixas_contagem[2]++;
            faixas_soma[2] += n;
        } else if (n < 1000) {
            faixas_contagem[4]++;
            faixas_soma[4] += n;
        } else if (n >= 1000) {
            faixas_contagem[3]++;
            faixas_soma[3] += n;
        }
    }

    if (contador > 0) {
        printf("\nA média dos valores inseridos é: %.2f\n", soma / contador);
        printf("O menor valor é: %.2f\n", menor);
        printf("O maior valor é: %.2f\n", maior);

        printf("\nValores pares:");
        if (indice_par == 0) {
            printf(" Nenhum\n");
        } else {
            for (int i = 0; i < indice_par; i++) {
                printf(" %d", pares[i]);
            }
            printf("\nA soma dos valores pares é: %.2f\n", soma_par);
        }

        printf("\nValores ímpares:");
        if (indice_impar == 0) {
            printf(" Nenhum\n");
        } else {
            for (int i = 0; i < indice_impar; i++) {
                printf(" %d", impares[i]);
            }
            printf("\nA soma dos valores ímpares é: %.2f\n", soma_impar);
        }
    } else {
        printf("Nenhum número foi inserido.\n");
    }

    printf("\nDistribuição por faixas:\n");
    for (int i = 0; i < 5; i++) {
        printf("Faixa %d - Total de elementos: %d, Total da faixa: %.2f\n", i + 1, faixas_contagem[i], faixas_soma[i]);
    }

    system("pause");
    return 0;
}
